### PR TITLE
chore: add tools-pkg

### PR DESCRIPTION
**Pull request by rumenov**

### CODE REVIEW

**Review by mraszyk — APPROVED** [review]

File: rs/backup/BUILD.bazel [hunk 1]
```diff
@@ -43,6 +41,10 @@ rust_library(
     crate_name = "ic_backup",
     proc_macro_deps = MACRO_DEPENDENCIES,
     version = "1.0.2",
+    visibility = [
+        "//rs:system-tests-pkg",
+        "//rs:tools-pkg",
```
Comment (kpop-dfinity) [inline rs/backup/BUILD.bazel:46]: I might be wrong but I don't think this target has to be visible to other packages (except the system tests)

```suggestion
        "//rs:system-tests-pkg",
```

**Review by kpop-dfinity — APPROVED** [review]

**Review by tthebst — APPROVED** [review]

File: rs/BUILD.bazel [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+#               |          \         |
+#               |         release    |
+#               |          |     \   |
+#               |          |     tools ( + integrations)
+#               |          |         |
+#               |          |         |
+#               |          |         |
+#               ======ic-protocol=====
+#               |          |         |
+#               BN     replica   dapps
```
Comment (basvandijk) [inline rs/BUILD.bazel:17]: What do the `vm` in those package mean? Maybe consider using the same name as the package_groups.

**Review by basvandijk — APPROVED** [review]

File: rs/BUILD.bazel [hunk 3]
```diff
@@ -21,28 +37,44 @@ package_group(
...
+# Can depend on: all targets excluding the //rs:system-tests-pkg and //rs:ic-os-pkg package groups.
+#
+# Dependants: //rs:ic-os-pkg
+package_group(
+    name = "release-pkg",
+    packages = ["//publish/..."],
+)
+
+# Tools useful for local dev., operational tasks, debugging, etc.
+# Can depend on: TODO: explain when replica-pkg is added
```
Comment (nmattia) [inline rs/BUILD.bazel:67]: leftover TODO

File: rs/pocket_ic_server/BUILD.bazel [hunk 4]
```diff
@@ -119,6 +117,8 @@ rust_binary(
     testonly = True,
     srcs = ["src/main.rs"],
     proc_macro_deps = MACRO_DEPENDENCIES,
+    # TODO: restrict the visibility
```
Comment (nmattia) [inline rs/pocket_ic_server/BUILD.bazel:120]: leftover TODO

**Review by nmattia — COMMENTED** [review]
There's a couple TODOs left, can you either resolve these in the PR or track them elsewhere?

File: rs/BUILD.bazel (hunk 2 above)
Comment (rumenov) [reply rs/BUILD.bazel:17]: yes, I will rename the packages in follow-up MRs. 

I guess I wanted to emphasize which packages deal with VMs and which should not.

File: rs/pocket_ic_server/BUILD.bazel (hunk 4 above)
Comment (rumenov) [reply rs/pocket_ic_server/BUILD.bazel:120]: This I will leave as a reminder for me for now.

Comment (nmattia) [reply rs/pocket_ic_server/BUILD.bazel:120]: @rumenov can you clarify the comment then? it's not clear to me why the visibility should/could be restricted, and why it's not done now

Comment (schneiderstefan) [reply rs/pocket_ic_server/BUILD.bazel:120]: I recommend making a ticket and mention the ticket number.

**Review by schneiderstefan — APPROVED** [review]

File: publish/binaries/BUILD.bazel [hunk 5]
```diff
@@ -2,6 +2,8 @@ load("//bazel:defs.bzl", "gzip_compress")
 load("//gitlab-ci/src/artifacts:upload.bzl", "upload_artifacts")
 load("//publish:defs.bzl", "release_nostrip_binary", "release_strip_binary", "release_strip_binary_test")
 
+package(default_visibility = ["//rs:ic-os-pkg"])
```
Comment (daniel-wong-dfinity-org) [inline publish/binaries/BUILD.bazel:5]: 😄

File: rs/registry/regedit/BUILD.bazel [hunk 6]
```diff
@@ -1,7 +1,5 @@
 load("@rules_rust//rust:defs.bzl", "rust_binary", "rust_library", "rust_test")
 
-package(default_visibility = ["//visibility:public"])
-
```
Comment (daniel-wong-dfinity-org) [inline rs/registry/regedit/BUILD.bazel:4]: 😄

**Review by daniel-wong-dfinity-org — APPROVED** [review]
Changes in the registry dir LGTM.

**Review by fspreiss — APPROVED** [review]

**Pull request merged**